torchaudio/__init__.py

```diff
@@ -125,85 +148,26 @@ def save(filepath, src, sample_rate, precision=16, channels_first=True):
         sample_rate (int): An integer which is the sample rate of the
             audio (as listed in the metadata of the file)
         precision (int): Bit precision (Default: ``16``)
-        channels_first (bool): Set channels first or length first in result. (
-            Default: ``True``)
     """
-    si = sox_signalinfo_t()
-    ch_idx = 0 if channels_first else 1
-    si.rate = sample_rate
-    si.channels = 1 if src.dim() == 1 else src.size(ch_idx)
-    si.length = src.numel()
-    si.precision = precision
-    return save_encinfo(filepath, src, channels_first, si)
-
-
-def save_encinfo(filepath,
-                 src,
-                 channels_first=True,
-                 signalinfo=None,
-                 encodinginfo=None,
-                 filetype=None):
-    r"""Saves a tensor of an audio signal to disk as a standard format like mp3, wav, etc.
+    if get_audio_backend() == "sox":
+        from torchaudio import sox_backend
```

Comment:
I'd carefully make sure that these repeated imports aren't expensive due to some kind of initialization code in sox.

Reply:
or soundfile for that matter

Reply:
One way of avoiding repeated import is to import at the beginning and catch import errors, as done in [vision](https://github.com/pytorch/vision/blob/master/torchvision/transforms/transforms.py#L7). However, local test on my mac don't seem to see a cost to repeated import and seem to properly fetch the cached version:
```
In [1]: %timeit import torchaudio                                                                  
The slowest run took 17.73 times longer than the fastest. This could mean that an intermediate result is being cached.
637 ns ± 1.07 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [2]: %timeit import _torch_sox                                                                  
76.3 ns ± 1.84 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```